packages/squiggle-lang/__tests__/Reducer/Reducer_Peggy/Reducer_Peggy_Parse_type_test.res

```diff
@@ -0,0 +1,79 @@
+open Jest
+open Reducer_Peggy_TestHelpers
+
+describe("Peggy parse type", () => {
+  describe("type of", () => {
+    testParse("p: number", "{(::$_typeOf_$ :p #number)}")
+  })
+  describe("type alias", () => {
+    testParse("type index=number", "{(::$_typeAlias_$ #index #number)}")
+  })
+  describe("type or", () => {
+    testParse(
+      "answer: number|string",
+      "{(::$_typeOf_$ :answer (::$_typeOr_$ (::$_constructArray_$ (#number #string))))}",
+    )
+  })
+  describe("type function", () => {
+    testParse(
+      "f: number=>number=>number",
```

Comment:
Would it be much harder to use something more like the typescript/rescript notation?
```
(number,number)=>number ? 
```
Happy to add it in a different story later on.

https://www.typescriptlang.org/docs/handbook/2/functions.html

I like the haskell/ocaml syntax a lot, but am afraid it would be more confusing to our user base.

Reply:
Yes, I can do that in standalone declarations...